libs/core/src/main/java/org/elasticsearch/common/io/TarEntry.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.io;
+
+import java.util.Objects;
+
+public class TarEntry {
```

Comment:
Maybe we should use a library that provides tar ball support? Like plexus-archiver or Apache commons-compress?

Reply:
I thought about it but both libraries adds quite big footprint (common-cmpress over 600kB, plexus over 200kB) just for this one use case. 
I think tar is simple enough (at least for basic cases but I don't expect anything crazy from infra) to parse it by ourselves. That said I can reconsider if you are really against creating our own solution here.

Reply:
I just realized that `plexus-archiver` actually depends on `common-compress`. I agree it is a lot of additional bytes for just for extracting a tarball and the code itself isn't complex. On the hand, about half of this PR is about adding support for extracting a tarball and comparing the size of all our other dependencies (all jdk libraries and the jdk) I think about 600kb for common-compress is just a small part our total distribution.

I did find a lightweight library called `jtar` that adds support for just tar files, but that project doesn't seem very active and maintained by basically a single developer.

I think I'm ok with maintaining this tar code ourselves, but it should be moved to ingest-geoipv2 module and explained that it is not a general tar library and meant to be used for extracting the tar files provided by the infra service.

Reply:
I moved it to geoip module and made it simpler by skipping everything that is not needed for our use case.